batteryChecker.c

```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include "batteryChecker.h"
+#include "test_batteryChecker.h"
+
+/***********Global Variable Declaration***********/
+bool	 batteryWarning;
+char*  	 batteryParameterString[] = {"Temperature", "State_of_Charge", "Charge_Rate"};
+/************************************************/
+
+void  batteryParameterCheck(char* parameterString, int  parameterValue, int maxLimit, int minLimit)
+{
+	if(parameterValue < minLimit || parameterValue > maxLimit)
+	{
+		printf("%s out of range!\n",parameterString);
+		batteryWarning = true;
+	}
+}
+
+bool batteryIsOk(struct batteryParameter batteryParam)
+{
+
+	batteryWarning = false;
```

Comment:
Usage of global variables is convenient but very vulnerable to errors. State management of global variables is error prone.
Here function return value and local variable would be more beneficial from a maintenance point of view.